src/WebJobs.Extensions.DurableTask/Listener/OutOfProcOrchestrationShim.cs

```diff
@@ -197,7 +197,12 @@ private async Task ProcessAsyncActionsV2(AsyncAction[] actions)
         {
             foreach (AsyncAction action in actions)
             {
-                await this.InvokeAPIFromAction(action);
+                // Before awaiting the task, we wrap it in a `WhenAny` call
+                // to protect against possible exceptions. Exception handling
+                // and exception throwing is managed in the OOProc SDKs and
+                // should not affect / interrupt this replay loop
+                var task = this.InvokeAPIFromAction(action);
```

Comment:
I'm willing to block the PR on this because this is extremely roundabout way to swallow an exception, and very confusing. +1 to doing try/catch instead.

Reply:
Fair enough, I only restored the `Task.WhenAny` because that recovers the "old and tested" way in which we swallowed these exceptions. A part of me wanted to bring it back in case there were even **more** **unknown** reasons why `WhenAny` was chosen in the past, but I agree that a try-catch is better for this particular situation.

Happy to change this

Reply:
`Task.WhenAny` was used to work for the case when we received multiple actions in an array from a .All() or .Any() call from the out-of-proc layer.

However, that clearly was incorrect for the .All() case, which lead to nondeterminism problems occasionally, as well as  issues with nested compound tasks.

There is no reason to use Task.WhenAny() now that we properly translate the tasks.